File: iosApp/iosApp/presentationLayer/pages/settings/viewModels/SettingsViewState.swift
enum SettingsViewState {

    case initial(
        accountShortCodeForShareTitle: String,
        accountsSharedWithMeTitle: String,
        accountsSharedWithMeSubtitle: String
    )

    case loading

    case success(
        accountShortCodeForShareTitle: String,
        accountsSharedWithMeTitle: String,
        accountsSharedWithMeSubtitle: String
    )
    
    case error(
        message: String,
        retryAction: (() -> Void)?
    )
}

File: androidApp/src/main/java/dev/haroldjose/familysharedlist/android/presentationLayer/pages/settings/viewmodels/SettingsViewState.kt
package dev.haroldjose.familysharedlist.android.presentationLayer.pages.settings.viewmodels

sealed class SettingsViewState {

    class Initial(
        val accountShortCodeForShareTitle: String,
        val accountsSharedWithMeTitle: String,
        val accountsSharedWithMeSubtitle: String
    ) : SettingsViewState()
    object Loading: SettingsViewState()
    class Success(
        val accountShortCodeForShareTitle: String,
        val accountsSharedWithMeTitle: String,
        val accountsSharedWithMeSubtitle: String
    ) : SettingsViewState()
    data class Error(
        val message: String,
        val retryAction: (() -> Unit)? = null
    ) : SettingsViewState()
}